src/main/java/ui/BoardNameDialog.java

```diff
@@ -0,0 +1,150 @@
+package ui;
+
+import java.util.ArrayList;
+
+import prefs.Preferences;
+import javafx.application.Platform;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.control.ButtonBar;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.Dialog;
+import javafx.scene.control.Button;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+import javafx.scene.text.Text;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+
+public class BoardNameDialog extends Dialog<String> {
+    
+    Preferences prefs;
+    TextField nameField;
+    Text prompt;
+    Text errorText;
+    Button submitButton;
+
+    private ArrayList<String> invalidNames = new ArrayList<>(); 
```

Comment:
A better data structure for this is a Set? Also it can be static, and you can use Collections.unmodifiableSet to freeze it.